Parser/spark.py

```diff
@@ -825,15 +830,15 @@ def resolve(self, list):
 
 def _dump(tokens, sets, states):
     for i in range(len(sets)):
-        print 'set', i
+        print('set', i)
         for item in sets[i]:
-            print '\t', item
+            print('\t', item)
             for (lhs, rhs), pos in states[item[0]].items:
-                print '\t\t', lhs, '::=',
-                print string.join(rhs[:pos]),
-                print '.',
-                print string.join(rhs[pos:])
+                 print('\t\t', lhs, '::=', end='')
```

Comment:
`end=' '`

But I think it would be better to use just a single multiline `print()`

Reply:
Yeah. My going principle was as few changes as possible – to make it easier to verify correctness.